src/mpid/ch4/generic/am/mpidig_am.h

```diff
@@ -11,9 +11,37 @@
 enum {
     MPIDIG_SEND = 0,            /* Eager send */
```

Comment:
One issue with this design is that the `MPIDIG_SEND` now has double meanings. It stands for "regular send message (irrespective to message length)" at some places and "EAGER SEND" at other places.

Maybe I should create `MPIDIG_SEND_EAGER` for the eager protocol related stuff (callbacks, etc.) and kept `MPIDIG_SEND` as general send message (just like `MPIDIG_SSEND_REQ` for ssend). In this way , we have a clear separate on which part is about the AM message, and which part is for the data transmissioin protocols (EAGER/PIPELINE/RDMA_READ).